### PR TITLE
Cleanup the native globalization code

### PR DESCRIPTION
**Pull request by tarekgh**
After we merged the globalization code used for Windows and Linux, this change is cleaning up the native VM globalization code

### CODE REVIEW

**Comment by tarekgh** [comment]
@jkotas this is a follow up cleanup after the globalization code merge done. 

cc @danmosemsft

**Comment by tarekgh** [comment]
@dotnet-bot help

**Comment by tarekgh** [comment]
@dotnet-bot test Ubuntu x64 corefx_baseline
@dotnet-bot test Windows_NT x64 corefx_baseline

File: src/classlibnative/inc/nlsinfo.h [hunk 1]
```diff
@@ -145,53 +86,7 @@ class COMNlsInfo {
 
     static void QCALLTYPE nativeNormalizationInitNormalization(int NormForm, BYTE* pTableData);
```
Comment (jkotas) [inline src/classlibnative/inc/nlsinfo.h]: Why do we need these normalization fcalls? I though that you have ported over the implementation that just PInvokes directly.

Comment (tarekgh) [inline src/classlibnative/inc/nlsinfo.h]: my changes was mainly to kill the duplication in the globalization code (i.e. get rid of the files under corefx folder). normalization stuff is not part of this, it is  used by string/System.Text. we can do your suggestion later.

Comment (tarekgh) [inline src/classlibnative/inc/nlsinfo.h]: I'll look more at that.

Comment (tarekgh) [inline src/classlibnative/inc/nlsinfo.h]: @jkotas we have some special case for normalization when we run on pre-Windows 8 (i.e. Win7). we carry the normalization data and use it when running on Windows 7. the reason is the normalization data on Windows 7 was not that good. if we are willing to scarify that when running on Windows 7, I can get rid of the fcalls along with the data too which will save us some more size.

Comment (jkotas) [inline src/classlibnative/inc/nlsinfo.h]: ok, we can leave it as is then ... it may be useful to add this as a comment.

Comment (tarekgh) [inline src/classlibnative/inc/nlsinfo.h]: The code using it is kind of clear about that:

https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Text/Normalization.Windows.cs#L48

but I'll add a comment anyway

Comment (jkotas) [inline src/classlibnative/inc/nlsinfo.h]: Actually, IsWindows8OrAbove is unconditionally returning true and we are not carrying the resources that would be required to make it work on pre-Win8 machines. We have shipped 1.0/1.1 that way. So all this is unreachable code.

Comment (tarekgh) [inline src/classlibnative/inc/nlsinfo.h]: Interesting. I can do the changes in other PR  then. ok?

Comment (jkotas) [inline src/classlibnative/inc/nlsinfo.h]: ok

**Pull request merged**